### PR TITLE
Refactor "unsubscribe" to reduce cognitive complexity

### PR DESCRIPTION
**Pull request by lucasrmendonca**
### Description, Motivation and Context
Refactor "unsubscribe" to reduce cognitive complexity

### What kind of change does this PR introduce?

- [x] Refactoring

### Checklist:
- [x] My code follows the code style of this project.
- [ ] I have added tests to cover my changes.
- [x] Overall test coverage is not decreased.
- [x] All new and existing tests passed.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/wampy.js [hunk 1]
```diff
@@ -1767,57 +1767,45 @@ class Wampy {
...
-            if (typeof onEvent === 'function') {
-                const i = sub.callbacks.indexOf(onEvent);
-                if (i >= 0) {
-                    sub.callbacks.splice(i, 1);
-                }
-            } else {
-                sub.callbacks = [];
-            }
+        subscription.callbacks = typeof onEvent === 'function'
+            ? subscription.callbacks = subscription.callbacks.filter((callback) => callback !== onEvent)
```
Comment (KSDaemon) [inline src/wampy.js:1790]:
```suggestion
            ? subscription.callbacks.filter((callback) => callback !== onEvent)
```

Comment (KSDaemon) [reply src/wampy.js:1790]: Seems that's not needed.

**Review by KSDaemon — COMMENTED** [review]
👍🏻 Just one small comment

File: src/wampy.js (hunk 1 above)
Comment (lucasrmendonca) [reply src/wampy.js:1790]: The previous solution was using Array.splice() to mutate the original array, however Array.filter() method doesn't mutate the original array, it creates a [shallow copy](https://developer.mozilla.org/en-US/docs/Glossary/Shallow_copy) of it. So I had to add this in order to keep mutating the `subscription.callbacks` array, so it has the same behavior as before.

**Comment by KSDaemon** [comment]
👍🏻 LGTM! Merging!

**Pull request merged**